gobblin-service/src/main/java/org/apache/gobblin/service/monitoring/FsJobStatusRetriever.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.gobblin.service.monitoring;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+
+import com.google.common.base.Joiner;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Predicate;
+import com.google.common.base.Splitter;
+import com.google.common.collect.Iterators;
+import com.typesafe.config.Config;
+
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+
+import org.apache.gobblin.configuration.State;
+import org.apache.gobblin.metastore.FileContextBasedFsStateStore;
+import org.apache.gobblin.metastore.FileContextBasedFsStateStoreFactory;
+import org.apache.gobblin.metastore.FsStateStore;
+import org.apache.gobblin.metrics.event.TimingEvent;
+
+
+/**
+ * A FileSystem based implementation of {@link JobStatusRetriever}. This implementation stores the job statuses
+ * as {@link org.apache.gobblin.configuration.State} objects in a {@link FsStateStore}.
+ * The store name is set to flowGroup.flowName, while the table name is set to flowExecutionId.jobName.
+ */
+@Slf4j
+public class FsJobStatusRetriever extends JobStatusRetriever {
+  @Getter
+  private final FileContextBasedFsStateStore<State> stateStore;
+
+  public FsJobStatusRetriever(Config config) {
+    this.stateStore = (FileContextBasedFsStateStore<State>) new FileContextBasedFsStateStoreFactory().createStateStore(config, State.class);
+  }
+
+  @Override
+  public Iterator<JobStatus> getJobStatusesForFlowExecution(String flowName, String flowGroup, long flowExecutionId) {
+    Preconditions.checkArgument(flowName != null, "FlowName cannot be null");
+    Preconditions.checkArgument(flowGroup != null, "FlowGroup cannot be null");
+
+    Predicate<String> flowExecutionIdPredicate = input -> input.startsWith(String.valueOf(flowExecutionId));
```

Comment:
Should be `String.valueOf(flowExecutionId) + "."` to avoid an incorrect prefix match.